fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java

```diff
@@ -132,8 +149,140 @@ public void addNumOfConcurrentTasksByBeId(long beId) {
         }
     }
 
-    public void addRoutineLoadJob(RoutineLoadJob routineLoadJob) {
-        idToRoutineLoadJob.put(routineLoadJob.getId(), routineLoadJob);
+    public void addRoutineLoadJob(CreateRoutineLoadStmt createRoutineLoadStmt)
+            throws AnalysisException, DdlException, LoadException {
+        // check load auth
```

Comment:
I think it's better to move auth check to statement analysis. Then this class needs not to know about auth things

Reply:
Analysis is used to check `Semantics`， but auth is not belong to `Semantics`. So maybe put auth check in here is better?